airbyte-integrations/connectors/source-datascope/source_datascope/source.py

```diff
@@ -0,0 +1,160 @@
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from datetime import datetime, timedelta
+import logging
+import backoff
+
+import requests
+from requests.exceptions import ConnectionError
+from requests.structures import CaseInsensitiveDict
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.sources.streams.http.auth import NoAuth
+
+
+
+# Basic full refresh stream
+class DatascopeStream(HttpStream, ABC):
+
+    # TODO: Fill in the url base. Required.
+    url_base = "https://mydatascope.com/api/external/"
+
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
```

Comment:
it seems like the API has some pagination params `limit` and `offset`, how come we are not using them?

Reply:
it's not using limit and offset, it's ordered by updated_at and then changing the last updated_at param on the next request. It's that ok?

Reply:
Ah -- so there is no concept of pagination in the API? (for example what if there are 10,000 records in a response? do they all get returned in a single page/response?)

Reply:
Is there a limit of 200 per request but the offset is not numeric, is the greater updated_at date of the last request because the request are sort by updated_at. I don't know if I understood well the incremental request. Do you think this is ok?

Reply:
Is sending duplicated data some time, could be related with this? In incremental Deduped + History it works perfect,   but if I put full refresh some data is duplicated. Any idea how to fix this?

Reply:
@agrass isn't duplication expected in full refresh mode? full refresh just means "read all the data every time"

Reply:
@sherifnada You are right sorry, I mean is duplicating on "Incremental - Append" and working well with deduped.

Reply:
@agrass duplication can sometimes happen in incremental append, it's expected behavior e.g: https://docs.airbyte.io/understanding-airbyte/connections/incremental-append#inclusive-cursors

Reply:
if you don't want this behavior then you can do a greater_than comparison on the cursor field and filter out records whose cursor is `<= cursor_field`

Reply:
Great, thanks @sherifnada. I understand, I was checking Intercom integration and is very similar of what I'm trying to do. I have some pending fixes on this PR, I will let you know when I'm ready with this points:

- [ ] Fix incremental duplicate data
- [ ] use standard API endpoints 
- [ ] Fix Auth token in params

Reply:
@agrass did you need assistance here?

Reply:
Yes, I already performed the pending updates, but need help to review if there something missing or anything I need to improve. Thanks for the help